### DISCUSSION TITLE
Implementing chat function using Kafka

### DISCUSSION THREAD

**Discussion by wijehyeon** [sproutt/eussya-eussya-api#132] [Ideas]
카프카 채팅기능 구현가능한가

- 기획상 채팅은 1:1채팅, 전체채팅기능이 있다. 

- 전체채팅을 카프카의 Pub-sub 으로 구현하더라도 메시지는 한 컨슈머에서만 전달
-> 전체채팅 구성원들을 kafka consumer group 설정하고 producer가 보낸 메시지를 가장 먼저 받은 consumer가 화면에 로드?

- 1:1 채팅에서도 순서보장을 위해 topic을 여러개, partition1로 해야할지, partition을 여러개로 만들고 해당 파티션만 긁어와야할지 
 여러모로  채팅에 적합하지 않은게 많다

**Comment by Bellroute** [comment]
저도 카프카 quick start 따라해보면서 채팅 기능 구현해봤었는데 
카프카는 분산처리에 더 특화된 것 같아서 단순히 채팅 메시지큐로만 사용하기엔 오버스펙인 것 같다는 생각이 들었습니다.
그래서 RabbitMQ 쓰기로 했는데 RabbitMQ는 아직 못알아봤습니다.
